homeassistant/components/xiaomi_ble/device_trigger.py

```diff
@@ -21,41 +21,117 @@
 from homeassistant.helpers.typing import ConfigType
 
 from .const import (
-    CONF_EVENT_PROPERTIES,
+    BUTTON_TYPE_A,
+    BUTTON_TYPE_B,
+    CONF_SUBTYPE,
     DOMAIN,
-    EVENT_PROPERTIES,
+    EVENT_CLASS,
+    EVENT_CLASS_BUTTON,
+    EVENT_CLASS_MOTION,
     EVENT_TYPE,
+    MOTION_DEVICE,
     XIAOMI_BLE_EVENT,
 )
 
-MOTION_DEVICE_TRIGGERS = [
-    {CONF_TYPE: "motion_detected", CONF_EVENT_PROPERTIES: None},
-]
+TRIGGERS_BY_TYPE = {
+    BUTTON_TYPE_A: ["press"],
+    BUTTON_TYPE_B: ["press", "double_press", "long_press"],
+    MOTION_DEVICE: ["motion_detected"],
+}
 
-MOTION_DEVICE_SCHEMA = DEVICE_TRIGGER_BASE_SCHEMA.extend(
-    {
-        vol.Required(CONF_TYPE): vol.In(
-            [trigger[CONF_TYPE] for trigger in MOTION_DEVICE_TRIGGERS]
-        ),
-        vol.Optional(CONF_EVENT_PROPERTIES): vol.In(
-            [trigger[CONF_EVENT_PROPERTIES] for trigger in MOTION_DEVICE_TRIGGERS]
-        ),
-    }
-)
+
+SCHEMA_BY_TYPE = {
+    BUTTON_TYPE_A: DEVICE_TRIGGER_BASE_SCHEMA.extend(
+        {
+            vol.Required(CONF_TYPE): vol.In([EVENT_CLASS_BUTTON]),
+            vol.Required(CONF_SUBTYPE): vol.In(TRIGGERS_BY_TYPE[BUTTON_TYPE_A]),
+        }
+    ),
+    BUTTON_TYPE_B: DEVICE_TRIGGER_BASE_SCHEMA.extend(
+        {
+            vol.Required(CONF_TYPE): vol.In([EVENT_CLASS_BUTTON]),
+            vol.Required(CONF_SUBTYPE): vol.In(TRIGGERS_BY_TYPE[BUTTON_TYPE_B]),
+        }
+    ),
+    MOTION_DEVICE: DEVICE_TRIGGER_BASE_SCHEMA.extend(
+        {
+            vol.Required(CONF_TYPE): vol.In([EVENT_CLASS_MOTION]),
+            vol.Required(CONF_SUBTYPE): vol.In(TRIGGERS_BY_TYPE[MOTION_DEVICE]),
+        }
+    ),
+}
 
 
 @dataclass
 class TriggerModelData:
     """Data class for trigger model data."""
 
-    triggers: list[dict[str, Any]]
     schema: vol.Schema
+    event_class: str
+    triggers: list[str]
 
 
 MODEL_DATA = {
+    "JTYJGD03MI": TriggerModelData(
+        schema=SCHEMA_BY_TYPE[BUTTON_TYPE_A],
+        event_class=EVENT_CLASS_BUTTON,
+        triggers=TRIGGERS_BY_TYPE[BUTTON_TYPE_A],
+    ),
```

Comment:
Maybe make TRIGGER_MODEL_A and TRIGGER_MODEL_B constants so you don't have to construct them over and over.

Ideally we name them something a bit more descriptive

Reply:
Good idea. Changed.